### PR TITLE
Add Support For AWS Dex Basic-Authentication

### LINKED ISSUE #1164: [AWS] Consider to use dex solution for basic aws manifest
https://github.com/kubeflow/manifests/blob/master/kfdef/source/master/kfctl_aws.yaml

We used a public application load balancer in the past. With some security concerns, we remove public one and ask user to use port-forward way to check dashboard. 

Recently we get a few asks from users on basic authentication, multi-user for [non-cognito manifest]((https://github.com/kubeflow/manifests/blob/master/kfdef/source/master/kfctl_aws_cognito.yaml) ).

Since community already has dex solution, I think we should consider to reuse this for  `kfctl_aws.yaml`

There're a few good things
- AWS folks can also help maintain and improve Dex solution. 
- Even Coginito setup is recommended and it's production grade solution, it's kind of complicated for users just like to try Kubeflow, for example, for workshop purpose. 
- two AWS manifest will be multi-user enabled and all our examples can be unified. 

Engineer stories
- Update kfctl_aws.yaml to include dex manifests
- Update kfctl to support new manifest orchestration
  - Add logic to handle basic auth
  - Handle dex installation in kfctl aws plugin 
- Still consider to use Application Load balancer instead of ELB (serviceType=Loadbalancer).

### PR DESCRIPTION
**Pull request by PatrickXYS**
**Which issue is resolved by this Pull Request:**
Resolves #1164 

**Description of your changes:**
Integrated Dex as basic authentication option in AWS Kubeflow.

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

### CODE REVIEW

File: dex-auth/dex-crds/base/params.env [hunk 1]
```diff
@@ -2,8 +2,8 @@
 dex_domain=dex.example.com
 # Set issuer to https if tls is enabled
 issuer=http://dex.example.com:32000
-static_email=admin@example.com
```
Comment (Jeffwan) [inline dex-auth/dex-crds/base/params.env:5]: This is a default change. User who not explicitly specify credentials may need it. Please don't touch it.

File: kfdef/kfctl_aws.yaml [hunk 2]
```diff
@@ -4,58 +4,50 @@ metadata:
...
         name: manifests
         path: istio/istio
     name: istio
-  - kustomizeConfig:
-      parameters:
-      - name: namespace
-        value: istio-system
-      repoRef:
-        name: manifests
-        path: istio/add-anonymous-user-filter
```
Comment (Jeffwan) [inline kfdef/kfctl_aws.yaml:45]: Good. We don't need this anymore

File: kfdef/kfctl_aws.yaml [hunk 3]
```diff
@@ -350,6 +387,16 @@ spec:
         name: manifests
         path: mpi-job/mpi-operator
     name: mpi-operator
+  - kustomizeConfig:
+      overlays:
```
Comment (Jeffwan) [inline kfdef/kfctl_aws.yaml:391]: Why do you need cognate overlay?

File: dex-auth/dex-crds/base/params.env (hunk 1 above)
Comment (Jeffwan) [reply dex-auth/dex-crds/base/params.env:5]: This is used by a few  kfdefs

File: kfdef/kfctl_aws.yaml (hunk 3 above)
Comment (PatrickXYS) [reply kfdef/kfctl_aws.yaml:391]: We don't need, copy & paste error here, fixed.

File: dex-auth/dex-crds/base/params.env (hunk 1 above)
Comment (PatrickXYS) [reply dex-auth/dex-crds/base/params.env]: Yeah I agree, fixed

File: kfdef/kfctl_aws.yaml [hunk 4]
```diff
@@ -374,9 +419,9 @@ spec:
     spec:
       auth:
         basicAuth:
-          password:
-            name: password
-          username: admin
+          password: $2y$12$ruoM7FqXrpVgaol44eRZW.4HWS8SAvg6KYVVSCIwKQPBmTpCm.EeO
```
Comment (Jeffwan) [inline kfdef/kfctl_aws.yaml:422]: Does user need to pass the encoded string?

Comment (PatrickXYS) [reply kfdef/kfctl_aws.yaml]: Right, fixed.

**Pull request merged**